### PR TITLE
use buffer-equals / buffer-reverse

### PR DESCRIPTION
**Pull request by dcousens**
Cons: 
* Added dependencies (`buffer-reverse`, `buffer-equals`)

Pros
* Actually tested (in the case of `buffer-equals`)
* Falls back to `Buffer.prototype.equals`,  inherent performance gain
* Less code to maintain
* Type checked

### CODE REVIEW

File: src/bufferutils.js [hunk 1]
```diff
@@ -184,7 +174,7 @@ function reverse (buffer) {
 }
 
 module.exports = {
-  equal: equal,
+  equal: require('buffer-equals'),
```
Comment (dcousens) [inline src/bufferutils.js]: Kept only to avoid a breaking change

**Comment by dcousens** [comment]
POST-ACKs welcome

**Comment by jprichardson** [comment]
ACK on `buffer-equals`. When we drop support for Node v0.10 we could use `Buffer.compare` though.

Not sure of the point of `buffer-reverse` though. Why not just `[].reverse.call(buffer)`?

**Comment by dcousens** [comment]
For the same reason we had a separate utility function for it in the first place.
We wanted to do a copy and reverse (immutability),  in *some* cases.

**Comment by dcousens** [comment]
Upon further review,  I don't think its necessary any more.

**Pull request merged**